packages/shared/src/hooks/usePostComment.ts

```diff
@@ -40,6 +40,45 @@ const getParentAndCurrentIndex = (
   return [parent, current];
 };
 
+interface SharedData {
+  editContent?: string;
+  editId?: string;
+}
+
+export const getParentComment = (
```

Comment:
This actually feels like we can uniform the return based on type maybe?
(optional comment)

Reply:
I think we already have a return type of `ParentComment`. Do you mean something else?

Reply:
Ah yes, I mean the both returns are basically the same was wondering if there was a way to uniform it into 1?

Reply:
Hmm. Not really sure how since each property that would be coming from either a post or comment was explicitly defined. Do you have any suggestions?

Reply:
Really not urgent, and not sure it's even better.
But something like

```
const commentProps = { author: name, content, etc } = comment;
const postProps = { source: name, title, etc } = post;
props = com ? comment ? commentProps : postProps;
```

As mentioned really not sure if it even makes sense, the fragment of code just looked as it could be optimized for this.

Reply:
I agree, the code block looks a bit not looking good. Perhaps we can go back here once we think of something to replace this entirely. WDYT?

Reply:
Yeah 100% I'll log it for future discussion :)